src/crimson/os/seastore/extentmap_manager.h

```diff
@@ -0,0 +1,119 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+
+#pragma once
+
+#include <iostream>
+
+#include <boost/intrusive_ptr.hpp>
+#include <boost/smart_ptr/intrusive_ref_counter.hpp>
+
+#include <seastar/core/future.hh>
+
+#include "common/Formatter.h"
+#include "include/ceph_assert.h"
+#include "include/buffer_fwd.h"
+
+#include "crimson/osd/exceptions.h"
+#include "crimson/os/seastore/seastore_types.h"
+#include "crimson/os/seastore/transaction_manager.h"
+
+#define PAGE_SIZE 4096
+#define EXTMAP_BLOCK_SIZE 4096
+
+namespace crimson::os::seastore {
+
+struct lext_map_val_t {
+  laddr_t laddr;
+  extent_len_t length = 0;
+
+  lext_map_val_t(
+    laddr_t laddr,
+    extent_len_t length)
+    : laddr(laddr), length(length) {}
+
+};
+class Extent
+{
+public:
+  objaddr_t logical_offset = 0;  //offset in object
+  laddr_t laddr;     // lextent start address aligned with block size.
+  extent_len_t length = 0;
+  explicit Extent(objaddr_t lo) : logical_offset(lo) { }
+
+  Extent(
+    objaddr_t lo,
+    laddr_t laddr,
+    extent_len_t length)
+    : logical_offset(lo), laddr(laddr), length(length) {}
+
+  void dump(ceph::Formatter* f) const;
+    friend bool operator<(const Extent &a, const Extent &b) {
+      return a.logical_offset < b.logical_offset;
+    }
+    friend bool operator>(const Extent &a, const Extent &b) {
+      return a.logical_offset > b.logical_offset;
+    }
+    friend bool operator==(const Extent &a, const Extent &b) {
+      return a.logical_offset == b.logical_offset;
+    }
+
+  ~Extent() {}
+};
+
+using ExtentRef = std::unique_ptr<Extent>;
```

Comment:
I don't see reason to heap allocate these things as they are small and simple -- just copy them.